src/ethereum_test_types/blob.py

```diff
@@ -0,0 +1,469 @@
+            )
+
+        assert Path(file_path).exists(), (
+            f"Tried to load static blob from file but {file_path} does not exist"
+        )
+
+        with open(file_path, "r", encoding="utf-8") as f:
+            json_str: str = f.read()
+
+        # reconstruct object
+        return Blob.model_validate_json(json_str)
+
+    def write_to_file(self):
+        """Take a blob object, serialize it and write it to disk as json."""
+        json_str = self.model_dump_json()
+        output_location = Blob.get_filepath(self.fork, self.seed, self.timestamp)
+
+        # warn if existing static_blob gets overwritten
+        if output_location.exists():
+            print(f"Blob {output_location} already exists. It will be overwritten.")
+
+        with open(output_location, "w", encoding="utf-8") as f:  # overwrite existing
+            f.write(json_str)
+
+    def verify_cell_kzg_proof_batch(self, cell_indices: list) -> bool:  # noqa: E501
+        """Check whether all cell proofs are valid and returns True only if that is the case."""
+        assert self.fork in ["osaka"], (
+            f"verify_cell_kzg_proof_batch() is not available for fork: {self.fork}"
+        )
+
+        assert self.cells is not None, ""  # TODO: write message
+
+        assert len(cell_indices) == len(self.cells), (
+            f"Cell Indices list (detected length {len(cell_indices)}) and Cell list (detected length {len(self.cells)}) should have same length."  # noqa: E501
+        )
+
+        # each cell refers to the same commitment
+        commitments: list[bytes] = [self.commitment] * len(cell_indices)
+
+        is_valid = ckzg.verify_cell_kzg_proof_batch(
+            commitments, cell_indices, self.cells, self.proof, TRUSTED_SETUP
+        )
+
+        return is_valid
+
+    def delete_cells_then_recover_them(self, deletion_indices: list[int]):
+        """
+        Simulate the cell recovery process in user-specified scenario.
+
+        Note: Requirement for successful reconstruction is having at least N of the 2N cells.
+
+        Theoretical Usage: You pass a cell list with to 128 elements to this function along with a list of deletion indices.
+        These cells will be deleted and then the ckzg recovery mechanism is used to repair the missing cells.
+        If no assertion is triggered the reconstruction was successful.
+        """  # noqa: E501
+        assert self.fork in ["osaka"], (
+            f"delete_cells_then_recover_them() is not available for fork: {self.fork}"
+        )
+
+        assert self.cells is not None, "..."  # TODO: write text
+
+        assert isinstance(self.proof, list), (
+            "This function only works when self.proof is a list, but it seems to be of type bytes (not a list)"  # noqa: E501
+        )
+
+        assert len(self.cells) == 128, (
+            f"You are supposed to pass a full cell list with 128 elements to this function, but got list of length {len(self.cells)}"  # noqa: E501
+        )
```

Comment:
```suggestion
        assert len(self.cells) == self.fork.blob_kzg_cell_proofs(), (
            f"You are supposed to pass a full cell list with {self.fork.blob_kzg_cell_proofs()} elements to this function, but got list of length {len(self.cells)}"  # noqa: E501
        )
```
Same for the rest of the function, it can be adapted in the same way.